### PR TITLE
Use 'vtex.responsive-values' to handle values passed to itemsPerPage prop

### PR DESCRIPTION
**Pull request by victorhmp**
#### What problem is this solving?

The `slider-layout` used its own logic to handle the changes in display size and device. This resulted in duplicated logic since we already have a solution for this in the form of `vtex.responsive-values`. With this PR we're cleaning things up by removing that unnecessary logic.

Also, with the most recent release of `vtex.responsive-values`, the support for media-queries was added, so we now support it in the `slider-layout` as well.

#### How to test it?

Go to [Workspace](https://victormiranda--storecomponents.myvtex.com/), and the sliders should be working fine. This is more of an internal change, so what you should be aware of is any strange behavior.

#### Related to / Depends on

This was motivated by https://github.com/vtex-apps/responsive-values/pull/9.

### CODE REVIEW

File: react/SliderLayout.tsx [hunk 1]
```diff
@@ -16,18 +16,25 @@ const SliderLayout: StorefrontFunctionComponent<SliderLayoutProps &
   fullWidth = true,
   arrowSize = 25,
   children,
+  itemsPerPage = {
+    desktop: 5,
+    tablet: 3,
+    phone: 1,
+  },
```
Comment (Klynger) [inline react/SliderLayout.tsx:23]: This might be a breaking change, no?

Comment (victorhmp) [reply react/SliderLayout.tsx:23]: Nope, I've just changed where this default value is defined, but this was already its value

**Comment by kaisermann** [comment]
@Klynger What did you find?

**Comment by Klynger** [comment]
> @Klynger What did you find?

I think he fixed already, but I talked with him on discord

**Comment by victorhmp** [comment]
> Something is wrong in small sizes

Should be fine now 🙏

**Comment by victorhmp** [comment]
> There are some lint issues in the mocks, could you fix it?

Yep, just did it ;)

**Pull request merged**